safe-global/safe-wallet-web
# feat: Edit proposer dialog [SW-391] [SW-396]

## What it solves

Resolves [SW-391](https://www.notion.so/safe-global/Edit-existing-proposer-12a8180fe573805397cfc85e292bc235)

## How this PR fixes it

- Creates a new Dialog called `EditProposerDialog`
- Refactors `DeleteProposerDialog` to follow the same pattern
- Refactors `AddProposer` to `UpsertProposer` to handle editing

## How to test it

1. Create a proposer
2. Go to the settings page and observe a new edit icon next to them
3. Press the button
4. Observe a new dialog to edit the proposer label

## Screenshots

<img width="634" alt="Screenshot 2024-10-28 at 11 18 53" src="https://github.com/user-attachments/assets/353b7f6a-b2fd-4c60-9b32-d3e11d45023e">


## Checklist
